server/services/user.service.js

```diff
@@ -541,39 +541,55 @@ export const getOneUser = async (userId) =>
     },
   ]);
 
-export const addDirector = async ({ directorInfo, userId }) => {
+export const editDirector = async ({ directorInfo, userId }) => {
   const user = await getUserById(userId).catch((error) => {
     throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
   });
 
+  const vendorIsForDirector = (director) =>
+    director._id.toString() === directorInfo.directorId.toString();
+
+  const validVendor = user.vendor.directors.some(vendorIsForDirector);
```

Comment:
`validVendor` is a confusing variable name since we are checking if the director is valid. You can choose a better name or use this:

```suggestion
  const directorIsValid = user.vendor.directors.some(vendorIsForDirector);
```